snuba/subscriptions/entity_subscriptions/factory.py

```diff
@@ -0,0 +1,118 @@
+from typing import Generator, Mapping, MutableMapping, Sequence, Type
+
+from snuba import settings
+from snuba.datasets.entities import EntityKey
+from snuba.subscriptions.entity_subscriptions.configuration.entity_subscription_builder import (
+    build_entity_subscription_from_config,
+)
+from snuba.subscriptions.entity_subscriptions.entity_subscription import (
+    EntitySubscription,
+)
+from snuba.utils.config_component_factory import ConfigComponentFactory
+from snuba.utils.serializable_exception import SerializableException
+
+
+class _EntitySubscriptionFactory(
+    ConfigComponentFactory[Type[EntitySubscription], EntityKey]
+):
+    def __init__(self) -> None:
+        self._entity_subscription_map: MutableMapping[
+            EntityKey, Type[EntitySubscription]
+        ] = {}
+        self._name_map: MutableMapping[Type[EntitySubscription], EntityKey] = {}
+        self.__initialize()
+
+    def __initialize(self) -> None:
+        from snuba.subscriptions.entity_subscriptions.entity_subscription import (
+            EntitySubscription,
+            EventsSubscription,
+            GenericMetricsDistributionsSubscription,
+            GenericMetricsSetsSubscription,
+            MetricsCountersSubscription,
+            MetricsSetsSubscription,
+            TransactionsSubscription,
+        )
+
+        entity_subscription_to_config_path_mapping: Mapping[EntityKey, str] = {
+            EntityKey.GENERIC_METRICS_SETS: "snuba/subscriptions/entity_subscriptions/generic_metrics/sets.yaml",
```

Comment:
This dictionary should be built automatically, see how the storages are loaded automatically: https://github.com/getsentry/snuba/blob/master/snuba/datasets/storages/factory.py#L97

Reply:
This might require some class (e.g. https://github.com/getsentry/snuba/pull/3109/) or relationship between (EntitySubscriptions and Entities) to map automatically imported file paths and EntityKeys. Perhaps I could create a ticket for this to discuss storing EntitySubscriptions onto Entities directly?

Reply:
Sure, that's fine.